python/ray/autoscaler/aws/config.py

```diff
@@ -332,15 +340,75 @@ def _get_vpc_id_or_die(config, subnet_id):
 
 
 def _get_security_group(config, vpc_id, group_name):
+    security_group = _get_security_groups(config, [vpc_id], [group_name])
+    return None if not security_group else security_group[0]
+
+
+def _get_security_groups(config, vpc_ids, group_names):
     ec2 = _resource("ec2", config)
     existing_groups = list(
         ec2.security_groups.filter(Filters=[{
             "Name": "vpc-id",
-            "Values": [vpc_id]
+            "Values": vpc_ids
         }]))
+    matching_groups = []
     for sg in existing_groups:
-        if sg.group_name == group_name:
-            return sg
+        if sg.group_name in group_names:
+            matching_groups.append(sg)
+    return matching_groups
+
+
+def _create_security_group(config, vpc_id, group_name):
+    client = _client("ec2", config)
+    client.create_security_group(
```

Comment:
Sounds good - I've added a basic implementation that will tag every newly created security group with the cluster name. 

It may be worth noting that this basic implementation is susceptible to a user later explicitly configuring this security group for use with a different ray cluster, in which case the tag would still identify the old cluster name. 

However, if we want to eventually make the assumption elsewhere in code that all security groups are tagged with the correct cluster name, then we would need several additional changes to ensure that all existing security group tags are kept up-to-date with the ray cluster that they're currently servicing. Immediate vs. eventual consistency becomes an interesting issue to grapple with here as well.

Reply:
This may be more work than we want to handle in this PR but I don't feel super good about knowingly giving wrong information. I'd almost prefer we tabled the tagging until we can make sure we can keep it accurate.

Reply:
Returning to tagging later to ensure that we can at least do an idempotent update of it every time the config is processed is probably a good idea. This is something we would be in a better position to do after disabling the config cache.

I expect that going further and trying to correct any case of tag drift outside of config bootstrapping gets even more complex, but it would be cool to explore available options a bit. I'll add a milestone to address this in https://github.com/ray-project/ray/issues/8420.

For now, I'll pull this out and place it on a separate fork.